### PR TITLE
Appealsv2 api integration mg

### PR DESCRIPTION
**Pull request by markgreenburg**
- Replace hard coded mock data return with actually querying the `appeals_v2` endpoint for appeals data.
- Add unit tests.

### CODE REVIEW

File: src/js/claims-status/actions/index.jsx [hunk 1]
```diff
@@ -84,16 +86,18 @@ export function fetchAppealsSuccess(response) {
...
-    // Fake the fetch by just returning a resolved promice with the object shape we expect
-    //  to get from the api.
-    return setTimeout(() => Promise.resolve(mockData)
-      .then((response) => dispatch(fetchAppealsSuccess(response)))
-      .catch(() => dispatch({ type: SET_APPEALS_UNAVAILABLE })), 4000);
+    return apiRequest(
+      '/appeals_v2',
+      null,
+      (appeals) => dispatch(fetchAppealsSuccess(appeals)),
+      (error) => {
```
Comment (markgreenburg) [inline src/js/claims-status/actions/index.jsx]: Might want to segment errors better, but should probably do that in a separate PR since there's work that needs to be done to gather error states and map them out to messages in the UI.

Comment (markgreenburg) [reply src/js/claims-status/actions/index.jsx]: Tracking in https://github.com/department-of-veterans-affairs/vets.gov-team/issues/8567

File: src/js/claims-status/actions/index.jsx [hunk 2]
```diff
@@ -84,16 +86,18 @@ export function fetchAppealsSuccess(response) {
...
+        Raven.captureException(`vets_appeals_v2_err_get_appeals ${error.message}`);
```
Comment (cvalarida) [inline src/js/claims-status/actions/index.jsx]: Should this be `Raven.captureMessage()`?

**Review by cvalarida — APPROVED** [review]
Looks great. Were you able to verify that the data structure received from the api is the same as we have in our mock data?

File: src/js/claims-status/actions/index.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/js/claims-status/actions/index.jsx]: 🤷‍♂️ wasn't sure what our standard usage is across the site; seems like we use both. I think you're right that captureMessage is probably more appropriate though, so updating.
